### DISCUSSION TITLE
Mismatch in CAN_DataFrame.ID between API and asammdf

### DISCUSSION THREAD

**Discussion by YoBruce45** [danielhrisca/asammdf#1078] [Q&A] [answered]
Hi community,
i am observing the following mismatch between the python API and the asammdf-tool (which i use to make some validity checks) 

-  I parse my mdf searching for IDs, using simply:
  ` print ("groupCounter " + str(groupCounter) + " : "+  str(group.loc[group.index[0],"CAN_DataFrame.CAN_DataFrame.ID"]))`

- in parallel I open the mdf with asammdf and display for the groups the "ID"column. 

Output in python: 
![image](https://github.com/user-attachments/assets/e72cd60a-769a-42bd-8162-858474ce2d99)

Output in asammdf:
![image](https://github.com/user-attachments/assets/e1482a8b-26cd-4458-9dcf-0690a1752dfe)

So as we see, sometimes the IDs do not match with each other, please see group 4.
Did somebody already face this ? Is it possibly a setting issue?
Thanks in advance =)

**Comment by danielhrisca** [comment, marked as answer]
The GUI tabular window has special handling for CAN IDs. Usually the ID is stored as uint32 in the measurement, and the most significant bit is used to mark if the message is extended or not. This bit is masked out before the ID is shown in the tabular window

![image](https://github.com/user-attachments/assets/0e6cc4a3-efd3-47d2-91c6-629cc3573396)

`mdf.get_group` does not mask the MSB

**Reply by YoBruce45** [reply]
Thank you very much for those precious hints ;)
